urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/jdbc/DBOpt.java

```diff
@@ -0,0 +1,61 @@
+package com.sinopec.jdbc;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Created with IntelliJ IDEA. User: gaochuanjun Date: 13-10-30 Time: 下午2:51 To
+ * change this template use File | Settings | File Templates.
+ */
+public class DBOpt {
+
+	private String ipAddress;
+
+	private String dataBaseName;
+
+	private String user;
+
+	private String password;
+
+	public DBOpt(String ipAddress, String user, String password, String dataBaseName) {
+		this.ipAddress = ipAddress;
+		this.user = user;
+		this.password = password;
+		this.dataBaseName = dataBaseName;
+	}
+
+	public String getDataBaseServerIP(String gathererNodeName) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.